packages/react-storage/src/components/StorageBrowser/createStorageBrowser/__tests__/createStorageBrowser.test-d.tsx

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable react/jsx-no-useless-fragment */
+import { expectAssignable, expectNotAssignable } from 'jest-tsd';
+import createStorageBrowser from '../createStorageBrowser';
+import { defaultActionConfigs } from '../../actions';
+import React from 'react';
+
+const { StorageBrowser, useAction, useView } = createStorageBrowser({
+  actions: {
+    default: {
+      ...defaultActionConfigs,
+    },
+    custom: {
+      myAction: {
+        handler: () => ({
+          result: Promise.resolve({ status: 'COMPLETE' }),
+        }),
+        viewName: 'MyActionView',
+        actionListItem: {
+          icon: 'info',
+          label: 'my action',
+        },
+      },
+    },
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  config: {} as any,
+});
+
+type StorageBrowserProps = Parameters<typeof StorageBrowser>[0];
+
+test('StorageBrowser component should support custom view and default view', () => {
+  expectAssignable<StorageBrowserProps>({
+    views: {
+      CopyView: () => <></>,
+      MyActionView: () => <></>,
+    },
+  });
+});
+
+test('StorageBrowser component should NOT support random views', () => {
+  expectNotAssignable<StorageBrowserProps>({
+    view: {
+      RandomView: () => <></>,
+    },
+  });
+});
+
+type StorageBrowserProviderProps = Parameters<
```

Comment:
Types should be declared above test code, also why not use the existing `StorageBrowserProviderProps` interface?